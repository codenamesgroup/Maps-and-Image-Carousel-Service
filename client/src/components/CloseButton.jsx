import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

class CloseButton extends React.PureComponent {
	static propTypes = {
		onClose: PropTypes.func,
	}

	static container = styled.div`
		text-align: center;
		color: lightgray;
		cursor: pointer;
		font-size: 16pt;
		user-select: none;

		&:hover {
			color: white;
		}
	`;

	render() {
		return (
			<CloseButton.container
				onClick={this.props.onClose}>
				exit âœ•
			</CloseButton.container>
		);
	}
}

export { CloseButton };