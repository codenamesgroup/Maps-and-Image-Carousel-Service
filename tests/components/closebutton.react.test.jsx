import React from 'react';
import { shallowToJson } from 'enzyme-to-json';

import { CloseButton } from './../../client/src/components/CloseButton.jsx';

describe('<CloseButton/>', () => {

	it('should render with long and lat props', () => {
		let component = shallow(<CloseButton/>);
		expect(shallowToJson(component)).toMatchSnapshot();
	});

	it('should render without props', () => {
		let component = shallow(<CloseButton/>);
		expect(shallowToJson(component)).toMatchSnapshot();
	});
});