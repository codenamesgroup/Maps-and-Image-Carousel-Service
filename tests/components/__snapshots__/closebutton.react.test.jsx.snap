// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseButton/> should render with long and lat props 1`] = `
<styled.div>
  exit ✕
</styled.div>
`;

exports[`<CloseButton/> should render without props 1`] = `
<styled.div>
  exit ✕
</styled.div>
`;
